### PR TITLE
CART-89 build: Use environment-module in scons_local

### PR DESCRIPTION
**Pull request by jolivier23**
Use environment module to load pkg-config for MPI versions

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: prereq_tools/base.py [hunk 1]
```diff
@@ -64,6 +64,151 @@
...
 
+class EnvironmentModule():
+    """Class for utilizing Modules component to load environment modules"""
+
+    def __init__(self):
+        """Load Modules for initializing envirables"""
+        self._module_func = self._init_module_func(lambda *x: False)
+        self._module_load = self._init_mpi_module()
+
+    def set_module_func(self, func):
```
Comment (frostedcmos) [inline prereq_tools/base.py]: where is this function used?

File: prereq_tools/base.py [hunk 2]
```diff
@@ -64,6 +64,151 @@
...
+        except subprocess.CalledProcessError:
+            return self._module_old
+
+        try:
+            exec(open(init_file).read())
+        except KeyError:
+            return default_func
+
+        # pylint: disable=undefined-variable
+        return module
```
Comment (frostedcmos) [inline prereq_tools/base.py]: what is this module var? a comment would help here

File: prereq_tools/base.py [hunk 3]
```diff
@@ -64,6 +64,151 @@
...
+        if mpi == "mpich":
+            load = mpich
+            unload = openmpi
+        else:
+            load = openmpi
+            unload = mpich
+
+        for to_load in load:
+            if self._module_func('is-loaded', to_load):
+                return True
```
Comment (frostedcmos) [inline prereq_tools/base.py]: is it safe to return if say mpich-x86_64 is loaded but not mpich?

File: prereq_tools/base.py [hunk 4]
```diff
@@ -64,6 +64,151 @@
...
+
+        for to_unload in unload:
+            if self._module_func('is-loaded', to_unload):
+                self._module_func('unload', to_unload)
+
+        for to_load in load:
+            if self._module_func('load', to_load):
+                print("Loaded %s" % to_load)
+                return True
```
Comment (frostedcmos) [inline prereq_tools/base.py]: why not just first unload everything in unload and then load only what you need?

File: prereq_tools/base.py (hunk 2 above)
Comment (jolivier23) [reply prereq_tools/base.py]: it gets defined in the exec call which essentially just prints a bunch of os.environ["..."] = ...

File: prereq_tools/base.py (hunk 3 above)
Comment (jolivier23) [reply prereq_tools/base.py]: yeah, those are common package names...we only want one of them

File: prereq_tools/base.py (hunk 4 above)
Comment (jolivier23) [reply prereq_tools/base.py]: could do that, I guess.  I'm sick of this patch right now

File: prereq_tools/base.py (hunk 1 above)
Comment (jolivier23) [reply prereq_tools/base.py]: Hmm, maybe not...it might be a holdover...I'll update the patch

File: prereq_tools/base.py (hunk 4 above)
Comment (jolivier23) [reply prereq_tools/base.py]: The main reason I did this is to preserve user environment

**Pull request merged**